docs/ELK-Stack.md

```diff
@@ -0,0 +1,368 @@
+# ELK Stack
+
+If you wish to run IntelMQ with ELK (Elasticsearch, Logstash, Kibana) it is entirely possible. This guide assumes the reader is familiar with basic configuration of ELK and does not aim to cover using ELK in general. It is based on the version 6.8.0 (ELK is a fast moving train therefore things might change). Assuming you have IntelMQ (and Redis) installation in place, lets dive in.
+
+## Configuring IntelMQ for Logstash
+
+In order to pass IntelMQ events to Logstash we are going to need an output queue, particulary orphaned queue (which simply means there is no IntelMQ bot processing events from this queue). However because IntelMQ system does not like to have an orphaned queue (and the IntelMQ Manager does not allow to create one), we will simply add a new bot at the end of the pipeline (or wherever you see fit) and set it's configuration `enabled` to `false`. The bot type does not matter as it's only a forever disabled dummy bot (it's recommended to choose an output bot, e.g. Rest API Output Bot).
+
+## Configuring Logstash
+
+Logstash defines pipeline as well. In the pipeline configuration of Logstash you need to specify where it should look for IntelMQ events, what to do with them and where to pass them.
+
+### Input
+
+This part describes how to receive data from Redis queue.
+See the example configuration and comments below:
+
+```
+input {
+  redis {
+    host => "10.10.10.10"
+    port => 6379
+    db => 2 
+    data_type => "list"
+    key => "restapi-output-queue"
+  }
+}
+```
+* `host` - Redis hostname or IP
+* `port` - Redis port
+* `db` - Redis db used (same value as `source_pipeline_db` from defaults.conf of IntelMQ)
+* `data_type` - data type is usually `list`
+* `key` - name of the queue (found in pipeline.conf of IntelMQ as source queue for our dummy bot)
+
+**Extra**
+
+You can also use syntax like this: `host => "${REDIS_HOST:10.10.10.10}"`
+
+The value will then be taken from environment variable `$REDIS_HOST`. If the environment variable is not defined then the default value of `10.10.10.10` will be used instead.
+
+### Filter (optional)
+
```

Comment:
we should try to keep all field modifications inside the IntelMQ ecosystem. We have an expert bot for the very purpose, https://github.com/certtools/intelmq/blob/develop/docs/Bots.md#format-field . The reason is that if the user has multiple outputs configured, he can apply changes at one place to have consistent output across the various output bots.

Reply:
I don't see the connection here. Format Fields Bot is used for for fields inside IntelMQ events. The note about environment variables is a different situation and concerns only the configuration of Logstash.

Reply:
The context is
Filter (optional)

i added the comment below that

Reply:
Ok, I didn't understand before. I agree, the filter part was only an example that it can be done, not necessarily that it should be done. I will change it. Thanks.